artemis-server/src/main/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImpl.java

```diff
@@ -958,6 +958,15 @@ public <T> T convert(Class<T> type, Object value) {
       updateApplyStatus(propsId, errors);
    }
 
+   private static boolean isClassProperty(String property) {
+      return property.endsWith(PROPERTY_CLASS_SUFFIX);
+   }
+
+   private static String extractPropertyClassName(String property) {
+      int propertyClassSuffixIndex = property.indexOf(PROPERTY_CLASS_SUFFIX);
+      return property.substring(0, propertyClassSuffixIndex);
+   }
```

Comment:
These two are a bit inconsistent, in that the first one only checks for _endsWith_, but the second extracts only up to the _first_ instance found, which could be different. Would be better if they both operated on the end basis for consistency (also saves looking for the location index of a value you can already calculate).